### PR TITLE
[Job API] Add more features tests to test_feature.py

### PR DESCRIPTION
**Pull request by willianrampazzo**
Add more features tests to test_feature.py. Also, add a command-line parameter to list features tested.

Signed-off-by: Willian Rampazzo <willianr@redhat.com>

### CODE REVIEW

File: selftests/job_api/test_features.py [hunk 1]
```diff
@@ -308,6 +316,21 @@ def test_check_output_file(self):
...
+    # Print features covered in this test
+    # ========================================================================
+    if args.features:
+        features = []
+        for suite in suites:
+            for variants in suite.config['run.dict_variants']:
+                features.append(variants['namespace'])
+
+        unique_features = sorted(set(features))
+        print('Total features covered: %i' %len(unique_features))
```
Comment (clebergnu) [inline selftests/job_api/test_features.py:417]: As a small tweak, I'd show this as:

```
Features covered (21):
job.output.loglevel
...
sysinfo.collect.enabled
```

Comment (willianrampazzo) [reply selftests/job_api/test_features.py]: That is much better. I have force pushed the change.

File: selftests/job_api/test_features.py [hunk 2]
```diff
@@ -84,20 +90,35 @@ def workdir_file_path(self):
         file_name = self.params.get('file')
         return os.path.join(self.latest_workdir, file_name)
 
+    def test_check_archive_file_exists(self):
```
Comment (clebergnu) [inline selftests/job_api/test_features.py:93]: I understand this check could *almost* reuse the "Test if a file was created", but the dynamic aspect of the logdir makes it hard. So this LGTM.

Comment (willianrampazzo) [reply selftests/job_api/test_features.py:93]: Right, tests that depend on variables from the job are tricky. I'm struggling while trying to merge tests as much as I can.

**Pull request merged**